webrtc.html

```diff
@@ -3124,30 +3124,51 @@ <h4>
                     </p>
                     <ol>
                       <li>
-                        <p>
-                          Parse the <var>url</var> using the generic URI syntax
-                          defined in [[!RFC3986]] and obtain the <var>scheme
-                          name</var>. If the parsing based on the syntax
-                          defined in [[!RFC3986]] fails, [= exception/throw =]
-                          a {{SyntaxError}}. If the <var>scheme name</var> is
-                          not implemented by the browser [= exception/throw =]
-                          a {{NotSupportedError}}. If <var>scheme name</var> is
-                          <code class="scheme">turn</code> or <code class=
-                          "scheme">turns</code>, and parsing the <var>url</var>
-                          using the syntax defined in [[!RFC7065]] fails, [=
-                          exception/throw =] a {{SyntaxError}}. If <var>scheme
-                          name</var> is <code class="scheme">stun</code> or
-                          <code class="scheme">stuns</code>, and parsing the
-                          <var>url</var> using the syntax defined in
-                          [[!RFC7064]] fails, [= exception/throw =] a
-                          {{SyntaxError}}.
-                        </p>
+                        <p>Let <var>parsedURL</var> be the result of
+                        <a data-cite="!url#concept-url-parser">parsing</a>
+                        <var>url</var> with <code>null</code>.</p>
+                      </li>
+                      <li>
+                        <p>If <var>parsedURL</var> is failure, then
+                        [= exception/throw =] a {{SyntaxError}}.
+                      </li>
+                      <li>
+                        <p>If <var>parsedURL</var>
+                        <a data-cite="!url#include-credentials">includes
+                        credentials</a>, then [= exception/throw =] a
+                        {{SyntaxError}}.
+                      </li>
+                      <li>
+                        <p>Let <var>scheme</var> be <var>parsedURL</var>'s
+                        <a data-cite="!url#concept-url-scheme">scheme</a>.</p>
+                      </li>
+                      <li>
+                        <p>If <var>scheme</var> is neither <code>stun</code>,
+                        <code>stuns</code>, <code>turn</code>, nor
+                        <code>turns</code>, then [= exception/throw =] a
+                        {{SyntaxError}}.</p>
+                      </li>
+                      <li>
+                        <p>If <var>scheme</var> is not implemented by the
+                        user agent, then [= exception/throw =]
+                        a {{NotSupportedError}}.
```

Comment:
This seems somewhat sketchy. Apart from it apparently allowing arbitrary schemes, this check would also pass for "file", "http", etc.

Reply:
The line above this one should ensure that by the time we get here, scheme is either `"stun"`, `"stuns"`, `"turn"`, or `"turns"`.

This line is here because not all browsers supported `"stuns"` and `"turns"` at the time.

Reply:
Ah of course, my bad!